src/scss/_mixins.scss

```diff
@@ -19,96 +19,136 @@
-@mixin _oTypographyFontSize($scale, $progressive-font-adjust: 1) {
+/// @param {Number} $scale - Number on scale the sizes are based on.
+/// @param {Number} $progressive-font-adjust - Multiplier for font-size adjustments, when adding styles for progressively loaded font.
+/// @param {String} $font [''] - The font to get the font size for, as fonts may have different scales. Uses the default scale if not specified.
+@mixin _oTypographyFontSize($scale, $progressive-font-adjust: 1, $font: '') {
 	@if type-of($scale) == map {
 		@each $breakpoint, $number in $scale {
 			@if $breakpoint == 'default' {
 				@if type-of($number) == list {
-					font-size: _oTypographyFontSizeFromScale(nth($number, 1), $progressive-font-adjust);
+					font-size: _oTypographyFontSizeFromScale(nth($number, 1), $progressive-font-adjust, $font);
 				} @else {
-					font-size: _oTypographyFontSizeFromScale($number, $progressive-font-adjust);
+					font-size: _oTypographyFontSizeFromScale($number, $progressive-font-adjust, $font);
 				}
 			} @else {
 				@include oGridRespondTo($breakpoint) {
 					@if type-of($number) == list {
-						font-size: _oTypographyFontSizeFromScale(nth($number, 1), $progressive-font-adjust);
+						font-size: _oTypographyFontSizeFromScale(nth($number, 1), $progressive-font-adjust, $font);
 					} @else {
-						font-size: _oTypographyFontSizeFromScale($number, $progressive-font-adjust);
+						font-size: _oTypographyFontSizeFromScale($number, $progressive-font-adjust, $font);
 					}
 				}
 			}
 
 		}
 	} @else {
-		font-size: _oTypographyFontSizeFromScale($scale, $progressive-font-adjust);
+		font-size: _oTypographyFontSizeFromScale($scale, $progressive-font-adjust, $font);
 	}
 }
 
-/// Outputs the the fallback font size only for progressively
-/// loaded fonts
+/// Set a custom font.
 ///
-/// @param {String} $font font to output fallback styles for
-/// @param {Number} $scale number on scale the sizes are based on
-@mixin _oTypographyProgressiveFontFallbackSize($font, $scale) {
-	@if $scale != false and map-has-key($_o-typography-progressive-font-fallbacks, $font) {
-		$font-config: map-get($_o-typography-progressive-font-fallbacks, $font);
-
-		.#{$o-typography-loading-prefix}-#{$font} & {
-			@include _oTypographyFontSize($scale, map-get($font-config, 'fallback-scale'));
-		}
+/// @example This example shows setting a custom font "MySansFont" as the "sans" font. We specify out custom font supports regular or bold variants.
+/// 	@include oTypographySetCustomFont($key: 'sans', $family: 'MySansFont, sans', $variants: (
+/// 		(weight: regular, style: normal),
+/// 		(weight: bold, style: normal)
+/// 	));
+///
+/// @param {String} $key - One of 'sans', 'serif', or 'display'.
+/// @param {String} $family - The font family to set.
+/// @param {Map} $variants - The variants (weight and style combinations) which are allowed in a nested map.
+@mixin oTypographySetCustomFont($key, $family, $variants) {
```

Comment:
A comment in the README is making me think — if we expect users to supply the correct weights and styles, do we have a solution in place in case they neglect to? e.g. `oTypographySansBold` won't return a heavy weight if the user didn't supply it and we didn't enforce it

Reply:
`oTypographySansBold` and other type mixins now delegate to `_oTypographyFor` which [checks the font variant](https://github.com/Financial-Times/o-typography/pull/163/files#diff-b2aec7b3cdbb70a5b21c2660f07c4c4cR21). So if a user sets a custom sans font with `oTypographySetCustomFont `, but doesn't say it supports a bold variant, `oTypographySansBold` will error.

Reply:
Worth noting `oTypographySansBold` will be deprecated in favour of `oTypographySans(opts: (weight: bold))`.

Reply:
👍